debugger/src/debugger/debug_entry.rs

```diff
@@ -93,9 +93,39 @@ impl Debugger {
                         Ok(DebuggerStatus::ContinueSession)
                     }
                     CoreStatus::Halted(_) => {
-                        session_data.poll_rtt(&self.config, debug_adapter);
-                        // No need to poll the target status if we know it is halted and waiting for us to do something.
-                        thread::sleep(Duration::from_millis(50)); // Small delay to reduce fast looping costs on the client
+                        // We check for RTT data the first time we have entered halted status, to ensure the buffers are drained.
+                        // After that, for as long as we remain in halted state, we don't need to check RTT again.
+                        // TODO: This only works for a single core, so until it can be redesigned, will use the first one configured.
+                        let check_rtt = if let Some(target_core_config) =
+                            self.config.core_configs.first_mut()
+                        {
+                            if let Ok(mut core_handle) =
+                                session_data.attach_core(target_core_config.core_index)
+                            {
+                                match core_handle.core.status() {
```

Comment:
Why do we poll the status again after we already know we are halted?

Reply:
Ah nvm I am stupid.

Reply:
We have to poll RTT at least once after halting, to ensure the buffer is drained. Otherwise valuable data might be stuck on the target until we start again.

I poll it a second time to compare against last_known_status, so that we can poll RTT once, and once only after a halt. 

Make sense?